pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -377,3 +395,63 @@ func indexedContainerEnv(envs []corev1.EnvVar) map[string]corev1.EnvVar {
 	}
 	return indexed
 }
+
+func hasSecurityContextChanged(current, desired *corev1.SecurityContext) bool {
+	if desired == nil {
+		return false
+	}
+
+	if current == nil {
+		return true
+	}
+
+	if desired.Capabilities != nil {
+		if current.Capabilities == nil {
+			return true
+		}
+		cmpCapabilities := cmpopts.SortSlices(func(a, b corev1.Capability) bool { return a < b })
+		if !cmp.Equal(desired.Capabilities.Add, current.Capabilities.Add, cmpCapabilities) {
+			return true
+		}
+
+		if !cmp.Equal(desired.Capabilities.Drop, current.Capabilities.Drop, cmpCapabilities) {
+			return true
+		}
+	}
```

Comment:
what if desired capabilities are `nil` but current are not?

Reply:
I've add some logic to handle the case for when the desired is `nil` but generally that won't be the case. I would not want the controller to set something to `nil` when it might be being set to some value but a webhook.